src/components/ArticleList.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div>
+    <div v-if="isLoading" class="article-preview">
+      Loading articles...
+    </div>
+    <rwv-article-preview
+      v-else
+      v-for="(article, index) in articles"
+      :article="article"
+      :key="article.title + index">
+    </rwv-article-preview>
+    <VPagination
+      :pages="pages"
+      :currentPage.sync="currentPage"
+    />
+  </div>
+</template>
+
+<script>
+  import RwvTag from '@/components/VTag'
+  import RwvArticlePreview from '@/components/VArticlePreview'
+  import VPagination from '@/components/VPagination'
+  import { FETCH_ARTICLES, FETCH_TAGS } from '@/store/actions.type'
+  import { GET_ARTICLE_COUNT, GET_ARTICLES, GET_ARTICLES_IS_LOADING } from '@/store/getters.type'
+
+  export default {
+    name: 'RwvArticleList',
+    components: {
+      RwvTag,
+      RwvArticlePreview,
+      VPagination
+    },
+    props: {
+      type: {
+        type: String,
+        required: false,
+        default: 'all'
+      },
+      author: {
+        type: String,
+        required: false
+      },
+      tag: {
+        type: String,
+        required: false
+      },
+      favorited: {
+        type: String,
+        required: false
+      }
+    },
+    data () {
+      return {
+        currentPage: 1
+      }
+    },
+    computed: {
+      listConfig () {
```

Comment:
Would a switch case be useful here?

Reply:
Nope, they are 3 diferent api parameters. So to keep the 3 parameters is useful